astropy/coordinates/representation.py

```diff
@@ -1193,16 +1489,17 @@ def scale_factors(self, omit_coslat=False):
 
     def represent_as(self, other_class):
         # Take a short cut if the other class is a spherical representation
-        if issubclass(other_class, PhysicsSphericalRepresentation):
-            return other_class(phi=self.lon, theta=90 * u.deg - self.lat,
-                               r=self.distance, copy=False)
-        elif issubclass(other_class, UnitSphericalRepresentation):
-            return other_class(lon=self.lon, lat=self.lat, copy=False)
-        else:
-            return super(SphericalRepresentation,
-                         self).represent_as(other_class)
+        if inspect.isclass(other_class):
```

Comment:
Thanks. I put in a solution that just skips the shortcut if a differential is attached, so it will work just not be as efficient when there are differentials. I also added a method that would help us avoid that / keep the optimization, but I'm running out of time, so in the interest of having it work I left TODO's.